### PR TITLE
Adding exported aksolve source code

### PR DESCRIPTION
**Pull request by mostroski**

### CODE REVIEW

File: aksolve/client/aksolve/conjugate_gradients.py [hunk 1]
```diff
@@ -0,0 +1,218 @@
...
+    b = matvec(x)
+    ans = cg(matvec, b, precon=P, verbose=True)
+    assert ak.all(ak.abs(ans[0] - x) < 1.0e-4)
+    assert ans[3] < 1.0e-6
+    assert ans[2] == 1
+    assert ans[1] == 3
+
+    print('##############')
+    print('# YOU PASSED #')
+    print('##############')
```
Comment (Ethan-DeBandi99) [inline aksolve/client/aksolve/conjugate_gradients.py:218]: This needs to be moved into a `test` directory (same level as client). Configure to use pytest.

File: aksolve/client/aksolve/minres.py [hunk 2]
```diff
@@ -0,0 +1,348 @@
...
+from typing import Optional, Tuple
+
+import numpy as np
+
+import arkouda as ak
+
+from aksolve.util import eye, EPS, FMAX, inner, norm, Operator
+
+
+def minres(
```
Comment (Ethan-DeBandi99) [inline aksolve/client/aksolve/minres.py:15]: Adding the comment here so It is easier overall. I am thinking we may want to break some of these calculations into sub-functions for readability. The functionality looks good to me, but it looks like there are sections that would easily break out.

File: aksolve/client/aksolve/minres.py [hunk 3]
```diff
@@ -0,0 +1,348 @@
...
+    b = matvec(x)
+    ans = minres(matvec, b, precon=P, verbose=True)
+    assert ak.all(ak.abs(ans[0] - x) < 1.0e-4)
+    assert ans[3] < 1.0e-6
+    assert ans[2] == 1
+    assert ans[1] == 3
+
+    print('##############')
+    print('# YOU PASSED #')
+    print('##############')
```
Comment (Ethan-DeBandi99) [inline aksolve/client/aksolve/minres.py:348]: Again, we need to move this to a `tests` directory and configure to use pytest.

File: aksolve/client/aksolve/symmlq.py [hunk 4]
```diff
@@ -0,0 +1,242 @@
+def symmlq(
```
Comment (Ethan-DeBandi99) [inline aksolve/client/aksolve/symmlq.py:1]: This function looks to have less natural breakpoints than the other, but I definitely think we could sub-function some elements out for readability here.

**Review by Ethan-DeBandi99 — CHANGES_REQUESTED** [review]
The functionality looks good to me. Mostly need to move testing into `test` directory and configure it to use pytest. A few functions could probably be split up into sub-functions for readability, but this is less important than test configuration.

File: aksolve/client/aksolve/__init__.py [hunk 5]
```diff
@@ -0,0 +1,2 @@
+from aksolve.util import *
```
Comment (mhmerrill) [inline aksolve/client/aksolve/__init__.py:1]: did you mean to import `symmlq.py` and `minres.py` stuff?

File: aksolve/test/aksolve_test.py [hunk 6]
```diff
@@ -0,0 +1,187 @@
...
+        P = lambda x: x * (1.0 / D)
+        b = matvec(x)
+        ans = aks.minres(matvec, b, precon=P, verbose=True)
+        self.assertTrue(ak.all(ak.abs(ans[0] - x) < 1.0e-4))
+        self.assertLess(ans[3], 1.0e-6)
+        self.assertEqual(ans[2], 1)
+        self.assertEqual(ans[1], 3)
+
+
+#TODO: finish converting these
```
Comment (mhmerrill) [inline aksolve/test/aksolve_test.py:126]: do these things still need to be worked?

**Review by mhmerrill — COMMENTED** [review]
i left comments

**Review by mhmerrill — APPROVED** [review]
looks ok to start, we will put in issues to cover changes that need to be made to work with the current version of Arkouda.

File: aksolve/client/aksolve/minres.py (hunk 2 above)
Comment (Ethan-DeBandi99) [reply aksolve/client/aksolve/minres.py:15]: Added Issue #35 to track

File: aksolve/test/aksolve_test.py (hunk 6 above)
Comment (Ethan-DeBandi99) [reply aksolve/test/aksolve_test.py:126]: Added issue #34 to track

**Pull request merged**